osu.Framework/Graphics/Containers/SearchContainer.cs

```diff
@@ -19,7 +19,13 @@ public class SearchContainer : SearchContainer<Drawable>
     /// Re-filtering will only be performed when the <see cref="SearchTerm"/> changes, or
     /// new items are added as direct children of this container.
     /// </summary>
-    /// <typeparam name="T"></typeparam>
+    /// <remarks>
+    /// <list type="bullet">
+    /// <item>Children which are searchable should be marked with the <see cref="IFilterable"/> interface. They do not need to be direct children.</item>
+    /// <item>Containers above <see cref="IFilterable"/>s which should be hidden only when all children are filtered away should also be marked with the <see cref="IFilterable"/> interface.</item>
+    /// <item>Containers which have the potential to hide children via external means should be <see cref="VisibilityContainer"/>s, in order to correctly exclude hidden child <see cref="IFilterable"/>s from search.</item>
+    /// </list>
+    /// </remarks>
```

Comment:
```suggestion
    /// <remarks>
    /// <list type="bullet">
    /// <item>Children which are searchable should be marked with the <see cref="IFilterable"/> interface. They do not need to be direct children to work (filtering will traverse the full drawable subtree).</item>
    /// <item>Marking a container (ie. a "group" or "section" that contains nested <see cref="IFilterable"/>s) as <see cref="IFilterable"/> will automatically keep it not-filtered as long as at least one nested item is not filtered away.</item>
    /// <item>Any <see cref="IFilterable"/>s which are contained in a <see cref="VisibilityContainer"/> that is hidden will be excluded from filtering. This can be used to exclude certain items from consideration via an external means (ie. items which have been collapsed), allowing group/sections to be correctly filtered away.</item>
    /// </list>
    /// </remarks>
```

Reworded to hopefully suck less (yes I know I wrote most of the previous version). Still not 100% on the last line.

Reply:
"items which have been collapsed" may feel misleading (in test scene, searching for a test scene already expands collapsed groups/folders). I feel it's not necessary to mention example of "external means".

Reply:
I've reworded slightly, see how that is now.

Reply:
Reads better :+1: